src/components/SignUp/index.jsx

```diff
@@ -1,13 +1,66 @@
 import React from "react";
 import { Container, Row, Col } from "react-bootstrap";
+import { useHistory } from "react-router-dom";
 import SignUpForm from "../SignUpForm";
 import womenSupportEachOther from "../../images/womenSupportEachOther.png";
 import googleIcon from "../../images/icons/googleIcon.svg";
 import twitterIcon from "../../images/icons/twitterIcon.svg";
 import facebookIcon from "../../images/icons/facebookIcon.svg";
+import firebase from "../../firebaseConfig";
 import "./index.scss";
 
 export default function SignUp() {
+  const history = useHistory();
+
+  const signUpFailed = (error) => {
+    alert(error);
+  };
+  const handleSignUp = React.useCallback(
+    async (data) => {
+      const { email, password } = data;
+
+      try {
+        await firebase.auth().createUserWithEmailAndPassword(email, password);
+        history.push("/");
+      } catch (error) {
+        if (error.code === "auth/email-already-in-use") {
+          return "auth/email-already-in-use";
```

Comment:
return error.code

Reply:
I thought that we might changed it to a custom error message. done!